text.py

```diff
@@ -380,8 +378,9 @@ def __init__(self, initial=None, goal=None, decoder=None):
 
     def actions(self, state):
         # Find the best
-        p, plainchar = max([(self.decoder.P1[c], c)
-                            for c in alphabet if c not in state])
+                            
+        search_list = [c for c in alphabet if c not in state]
+        plainchar = argmax(search_list, key=lambda c: self.decoder.P1[c])
```

Comment:
Did you mean self.decoder.P1.__getitem__
I got an error when I tried self.decoder.P1.get `AttributeError: 'UnigramTextModel' object has no attribute 'get'`